app/src/main/java/com/example/tasky/agenda/data/util/DateUtil.kt

```diff
@@ -1,14 +1,27 @@
+@file:SuppressLint("NewApi")
 package com.example.tasky.agenda.data.util
 
 import android.annotation.SuppressLint
 import java.time.Instant
+import java.time.LocalDate
 import java.time.LocalDateTime
+import java.time.LocalTime
 import java.time.ZoneId
 
-@SuppressLint("NewApi")
 fun Long.toCurrentTime(): String {
     return LocalDateTime.ofInstant(
         Instant.ofEpochMilli(this),
         ZoneId.systemDefault()
     ).toLocalTime().toString()
-}
\ No newline at end of file
+}
+
+
+fun LocalDate.asTodayStartLong() =
```

Comment:
I would rename this to
".toStartOfDayLong()"

Since you aren't doing it for Today only. You are accepting any date (It's an extension function for any LocalDate)